crates/bevy_math/src/bounding/raytest2d.rs

```diff
@@ -0,0 +1,98 @@
+use super::{Aabb2d, BoundingCircle, IntersectsVolume};
+use crate::{primitives::Direction2d, Ray2d, Vec2};
+
+/// A raycast intersection test for 2D bounding volumes
+pub struct RayTest2d {
+    /// The ray for the test
+    pub ray: Ray2d,
+    /// The maximum time of impact of the ray
+    pub max: f32,
+    /// The multiplicative inverse direction of the ray
+    dir_recip: Vec2,
+}
+
+impl RayTest2d {
+    /// Construct a [`RayTest2d`] from an origin, [`Direction2d`] and max time of impact.
+    pub fn new(origin: Vec2, direction: Direction2d, max: f32) -> Self {
+        Self::from_ray(Ray2d { origin, direction }, max)
+    }
+
+    /// Construct a [`RayTest2d`] from a [`Ray2d`] and max time of impact.
+    pub fn from_ray(ray: Ray2d, max: f32) -> Self {
+        Self {
+            ray,
+            dir_recip: ray.direction.recip(),
+            max,
+        }
+    }
+
+    /// Get the cached multiplicate inverse of the direction of the ray
+    pub fn dir_recip(&self) -> Vec2 {
+        self.dir_recip
+    }
```

Comment:
I would maybe rename `dir_recip` to `direction_recip` considering we have `direction` on the ray types. No strong feelings though